webhooks/webhooks.go

```diff
@@ -5,186 +5,92 @@ import (
 	XMLName xml.Name
 }
 
-// Account represents the account object sent in webhooks.
-type Account struct {
-	XMLName     xml.Name `xml:"account"`
-	Code        string   `xml:"account_code,omitempty"`
-	Username    string   `xml:"username,omitempty"`
-	Email       string   `xml:"email,omitempty"`
-	FirstName   string   `xml:"first_name,omitempty"`
-	LastName    string   `xml:"last_name,omitempty"`
-	CompanyName string   `xml:"company_name,omitempty"`
-	Phone       string   `xml:"phone,omitempty"`
-}
-
-// Transaction represents the transaction object sent in webhooks.
-type Transaction struct {
-	XMLName           xml.Name         `xml:"transaction"`
-	UUID              string           `xml:"id,omitempty"`
-	InvoiceNumber     int              `xml:"invoice_number,omitempty"`
-	SubscriptionUUID  string           `xml:"subscription_id,omitempty"`
-	Action            string           `xml:"action,omitempty"`
-	AmountInCents     int              `xml:"amount_in_cents,omitempty"`
-	Status            string           `xml:"status,omitempty"`
-	Message           string           `xml:"message,omitempty"`
-	GatewayErrorCodes string           `xml:"gateway_error_codes,omitempty"`
-	FailureType       string           `xml:"failure_type,omitempty"`
-	Reference         string           `xml:"reference,omitempty"`
-	Source            string           `xml:"source,omitempty"`
-	Test              recurly.NullBool `xml:"test,omitempty"`
-	Voidable          recurly.NullBool `xml:"voidable,omitempty"`
-	Refundable        recurly.NullBool `xml:"refundable,omitempty"`
+// Parse parses an incoming webhook and returns the notification.
+func Parse(r io.Reader) (interface{}, error) {
+	return parse(r, dst)
 }
 
-// Invoice represents the invoice object sent in webhooks.
-type Invoice struct {
-	XMLName             xml.Name         `xml:"invoice,omitempty"`
-	SubscriptionUUID    string           `xml:"subscription_id,omitempty"`
-	UUID                string           `xml:"uuid,omitempty"`
-	State               string           `xml:"state,omitempty"`
-	InvoiceNumberPrefix string           `xml:"invoice_number_prefix,omitempty"`
-	InvoiceNumber       int              `xml:"invoice_number,omitempty"`
-	PONumber            string           `xml:"po_number,omitempty"`
-	VATNumber           string           `xml:"vat_number,omitempty"`
-	TotalInCents        int              `xml:"total_in_cents,omitempty"`
-	Currency            string           `xml:"currency,omitempty"`
-	CreatedAt           recurly.NullTime `xml:"date,omitempty"`
-	ClosedAt            recurly.NullTime `xml:"closed_at,omitempty"`
-	NetTerms            recurly.NullInt  `xml:"net_terms,omitempty"`
-	CollectionMethod    string           `xml:"collection_method,omitempty"`
+// ParseOriginal parses an incoming webhook and returns the notification
+// for previous webhook notifications.
+// Will be deprecated after credit invoices feature is turned on.
+func ParseOriginal(r io.Reader) (interface{}, error) {
```

Comment:
Some of the constant names related to the credit invoices are `FooBarDeprecated`. Should we use the same naming convention here? `Parse` and `ParseDeprecated`?

Reply:
Yeah I like that.